src/main/java/com/artipie/repo/OrgLayout.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.repo;
+
+import com.artipie.RepoConfig;
+import com.artipie.Settings;
+import com.artipie.SliceFromConfig;
+import com.artipie.asto.Key;
+import com.artipie.http.Slice;
+import com.artipie.http.async.AsyncSlice;
+import com.artipie.http.rs.RsStatus;
+import com.artipie.http.rs.RsWithBody;
+import com.artipie.http.rs.RsWithStatus;
+import com.artipie.http.slice.SliceSimple;
+import com.jcabi.log.Logger;
+import io.vertx.reactivex.core.Vertx;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Pattern;
+import org.cactoos.scalar.Unchecked;
+
+/**
+ * Flat repositories layout.
+ * @see RepoLayout
+ * @since 0.4
+ * @checkstyle ReturnCountCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class OrgLayout implements RepoLayout {
```

Comment:
Except for one field `REPO_PREF`, how `OrgLayout` is different from `FlatLayout`? I could not find any differences.

Reply:
I think it is a code duplication which should be fixed.

Reply:
@Sammers21 `OrgLayout` reads two first part of URI to find repository config and to specify storage prefix, `FlatLayout` uses only first. But it can be refactored later.

Reply:
@g4s8, where is the puzzle for that?